ludwig/visualize.py

```diff
@@ -36,45 +36,88 @@
 
 
-def learning_curves(
-        training_statistics,
-        field,
-        model_names=None,
-        output_directory=None,
-        file_format='pdf',
-        **kwargs
-):
-    if len(training_statistics) < 1:
-        logging.error('No training_statistics provided')
+def load_data_for_viz(training_statistics,**kwargs):
+    """Load model training data in to json objects.
+
+    :param training_statistics: JSON file or list of json files containing the
+        model training stats.
+    :param kwargs: other arguments passed from the argparser.
+    :return training_statistics_per_model_name: List of training statistics l
+            oaded as json objects.
+    """
+    try:
+        training_statistics_per_model_name = [load_json(learning_stats_f)
+                                              for learning_stats_f in
+                                              training_statistics]
+    except (TypeError, AttributeError) as err:
+        logging.exception(
+            'Unable to open training statistics file {}!'.format(
+                training_statistics
+            )
+        )
         return
+    return training_statistics_per_model_name
 
-    filename_template = None
-    if output_directory:
-        filename_template = os.path.join(
-            output_directory,
-            'learning_curves_{}_{}.' + file_format
-        )
 
-    training_statistics_per_model_name = [load_json(learning_stats_f)
-                                          for learning_stats_f in
-                                          training_statistics]
+def validate_visualisation_prediction_field(
+        field,
+        training_statistics_per_model_name
+):
+    """Validate prediction field and return it as iterable.
 
+    :param field: field containing ground truth
+    :param training_statistics_per_model_name: list of per model train stats
+    :return fields: list of fields containing ground truth
+    """
     fields_set = set()
     for ls in training_statistics_per_model_name:
         for _, values in ls.items():
             for key in values:
                 fields_set.add(key)
-    fields = [field] if field is not None and len(field) > 0 else fields_set
+    fields = [field] if field in fields_set else fields_set
+    return fields
+
+def generate_filename_template_path(output_dir, filename_template):
+    """Ensure path to template file can be constructed given an output dir.
+
+    Create output directory if yet does exist.
+    :param output_dir: Directory that will contain the filename_template file
+    :param filename_template: name of the file template to be appended to the
+            filename template path
+    :return: path to filename template inside the output dir or None if the
+             output dir is None
+    """
+    if output_dir:
+        os.makedirs(output_dir, exist_ok=True)
+        return os.path.join(output_dir, filename_template)
+    return None
+
+def learning_curves(
```

Comment:
this looks fine for learning curves, I would just:
- rename `training_statistics_per_model_name` to something shorter, maybe `training_statistics_per_model`
- make it work for both lists and single values, like check if it is a list, if not do `training_statistics_per_model = [training_statistics_per_model]`
- the same thing applies to `model_names`, make it work both if it is a string or a list of strings
- the same for `field`, make it work for one single string or a list of strings

Reply:
Type checking is not very pythonic paradigm - “it’s easier to ask for forgiveness than permission”. But maybe I can write converter functions that do so 

Or we can ask  API users to pass them as iterable (list or tuple) so that the functionality continues to work - I am currently writing the integration tests in such way.

Let me know which one you prefer.

Reply:
From the developer experience point of view, if I have only one training statistics file and one algorithm name, I would prefer to call `learning_curves(training_stat, alg_name)` rather than `learning_curves([training_stat], [alg_name])`.
If there are better ways than checking for the type, that's great, you can figure it out :)